File: plugins/OrchidPosts/src/main/java/com/eden/orchid/posts/permalink/pathTypes/TagPathType.java
package com.eden.orchid.posts.permalink.pathTypes;

import com.eden.orchid.api.theme.pages.OrchidPage;
import com.eden.orchid.posts.pages.PostTagArchivePage;
import com.eden.orchid.posts.permalink.PermalinkPathType;

import javax.inject.Inject;

public class TagPathType extends PermalinkPathType {

    @Inject
    public TagPathType() {
        super(100);
    }

    @Override
    public boolean acceptsKey(OrchidPage page, String key) {
        return key.equals("tag");
    }

    @Override
    public String format(OrchidPage page, String key) {
        if(page instanceof PostTagArchivePage) {
            return ((PostTagArchivePage) page).getTag();
        }

        return null;
    }

}

File: plugins/OrchidPosts/src/main/kotlin/com/eden/orchid/posts/permalink/pathTypes/TagPathType.kt
package com.eden.orchid.posts.permalink.pathTypes

import com.eden.orchid.api.theme.pages.OrchidPage
import com.eden.orchid.posts.pages.PostTagArchivePage
import com.eden.orchid.posts.permalink.PermalinkPathType

import javax.inject.Inject

class TagPathType @Inject
constructor() : PermalinkPathType(100) {

    override fun acceptsKey(page: OrchidPage, key: String): Boolean {
        return key == "tag"
    }

    override fun format(page: OrchidPage, key: String): String? {
        return (page as? PostTagArchivePage)?.tag

    }

}

